src/main/drivers/accgyro_fake.c

```diff
@@ -28,26 +50,55 @@
 #include "accgyro.h"
 #include "accgyro_fake.h"
 
-
 static int16_t fakeGyroADC[XYZ_AXIS_COUNT];
 
+#if defined(SIMULATOR_BUILD) && defined(SIMULATOR_MULTITHREAD)
```

Comment:
My thinking as well. Some of the simulator code should become mainstream, rather than an add-on. As you say, there is a possible future where mutexes are used without the simulator.

See this discussion PR for changes to the fake acc and gyro:
https://github.com/betaflight/betaflight/pull/2849